src/control/server/mgmt_system.go

```diff
@@ -439,9 +439,11 @@ type (
 	systemRanksFunc func(context.Context, control.UnaryInvoker, *control.RanksReq) (*control.RanksResp, error)
 
 	fanoutRequest struct {
-		Method       systemRanksFunc
-		Hosts, Ranks string
-		Force        bool
+		Method     systemRanksFunc
+		Ranks      *system.RankSet
+		Force      bool
+		FullSystem bool
+		Resp       *fanoutResponse
```

Comment:
this is a bit weird having a response within a request

Reply:
I agree, this looks pretty weird to me. I get why it's there, after reading the code, but IMO it'd be better to pass around the absent hosts/ranks a different way. Replacing Resp here with the AbsentHosts/AbsentRanks variables directly would be clearer.

Reply:
OK, changed it so that `getFanoutReq()` returns the base response, which is then passed into `rpcFanout()` as a parameter.

The aim was to decouple the logic for creating the requests (which involves dealing with the absent ranks/hosts stuff) from the logic to actually perform the fanout.

To be honest, I'm not entirely sure what value the absent ranks/hosts stuff adds, but I was refraining from changing things too much in favor of trying to be focused on the immediate goal of determining when a system request is for the full system or a subset of ranks.

Reply:
the absent hosts/ranks code was added as there was a request to ensure that operations requested on entities that didn't exist were reported